src/java.net.http/share/classes/jdk/internal/net/http/Http2Connection.java

```diff
@@ -1204,6 +1208,21 @@ private void sendConnectionPreface() throws IOException {
         subscriber.onNext(List.of(EMPTY_TRIGGER));
     }
 
+    /**
+     * Called do get the initial stream after a connection upgrade.
+     * If the stream was cancelled, it might no longer be in the
+     * stream map. Therefore - we use the initialStream field
+     * instead, and reset it to null after returning it.
+     * @param <T> the response type
+     * @return the initial stream created during the upgrade.
+     */
+    @SuppressWarnings("unchecked")
+    <T> Stream<T> getInitialStream() {
```

Comment:
Given that this method returns and also updates the initial stream member field, the naming of this method is a bit odd. But I can't think of a better name, plus this is internal to the `jdk.internal.net.http` package and also has a comment which explains what it does, so I think this name is fine.

Reply:
Maybe I could change the name to "retrieveInitialStream()"... Would that be better?

Reply:
Hello Daniel, I don't have strong preference, `retrieveInitialStream` sounds OK. It's also OK if you would want to continue with the current name.